rdm/init_files/documents/software_plan.md

```diff
@@ -502,45 +528,49 @@ Specifications, on the other hand, should:
 
 Software requirements are often categorized as one of the following types [[62304:5.2.2 and 62304:5.2.3]]:
 
-a. Functional and capability requirements
+a. Functional and capability requirements [[62304:5.2.2.a]]
   - performance (e.g., purpose of software, timing requirements),
   - physical characteristics (e.g., code language, platform, operating system),
   - computing environment (e.g., hardware, memory size, processing unit, time zone, network infrastructure) under which the software is to perform, and
   - need for compatibility with upgrades or multiple SOUP or other device versions.
 
-b. Software system inputs and outputs
+b. Software system inputs and outputs [[62304:5.2.2.b]]
   - data characteristics (e.g., numerical, alpha-numeric, format) ranges,
   - limits, and
   - defaults.
 
-c. Interfaces between the software system and other systems
+c. Interfaces between the software system and other systems [[62304:5.2.2.c]]
 
-d. Software-driven alarms, warnings, and operator messages
+d. Software-driven alarms, warnings, and operator messages [[62304:5.2.2.d]]
 
-e. Security requirements
+e. Security requirements [[62304:5.2.2.e]]
   - those related to the compromise of sensitive information,
   - authentication,
   - authorization,
   - audit trail, and
   - communication integrity.
 
-f. Usability engineering requirements that are sensitive to human errors and training
+f. Usability engineering requirements that are sensitive to human errors and training [[62304:5.2.2.f]]
   - support for manual operations,
   - human-equipment interactions,
   - constraints on personnel, and
   - areas needing concentrated human attention.
 
-g. Data definitions and database requirements
+g. Data definitions and database requirements  [[62304:5.2.2.g]]
 
-h. Installation and acceptance requirements of the delivered medical device software at the operation and maintenance site or sites
+h. Installation and acceptance requirements of the delivered medical device software 
+at the operation and maintenance site or sites [[62304:5.2.2.h]]
 
-i. Requirements related to methods of operation and maintenance
+i. Requirements related to methods of operation and maintenance [[62304:5.2.2.i]]
 
-j. User documentation to be developed
+{# Keep this if meeting 62304 2006 AMD1 #}
```

Comment:
Ugh... I wonder if there is a way to avoid this sort of messiness.

I think it is ok if our default templates only support the newest FDA approved ISO standards, which in the case of 62304 if only the amended one.

See:

![Screen Shot 2020-08-20 at 2 46 05 PM](https://user-images.githubusercontent.com/1648754/90812321-ed09e380-e2f3-11ea-9d9a-6cedce7a4545.png)

You can search https://www.accessdata.fda.gov/scripts/cdrh/cfdocs/cfstandards/search.cfm for "62304" to see this.

The problem is some users may want to go back to the older templates (probably unlikely, but it could happen). In that case, someone may mistakenly think their passing audit is ok.

Perhaps we could avoid this by making "62304_AMD2015:5.2.2.j" and "62304_2006:5.2.2.j" ? Can you think of any other solution?

I think there is a fundamental underlying question regarding our "design goals" for RDM; do we want to provide really simple defaults? Or do we want to make the templates very flexible?

I tend to think simple defaults are more important than making the software work well for people who want the older standard. What do you think?

Reply:
Using the fancier reference tags will nicely flag things that need changing if the user switches standards. I will add that nuance. 

After that I like leaving both in with instructions to delete one of them. The document will be getting a lot of custom edits already. 

It is worth thinking about having something in `system.yml` that specifies the target standard and selects one over the other. I am inclined to say no to that for now.

Reply:
I have modified the document and the reference key. I went with `62304:5.2.2.j@62304_AMD2015` and `62304:5.2.2.j@62304_2006`. While a little redundant, it keeps the sectional sorting intact. We may need this later if we want to get fancy with an automated index.